xm-commons-domainevent/src/main/java/com/icthh/xm/commons/domainevent/config/WebApiSource.java

```diff
@@ -0,0 +1,118 @@
+            return;
+        }
+
+        XmAuthenticationContext auth = springSecurityXmAuthenticationContextHolder.getContext();
+        if (auth == null || auth.isAnonymous()) {
+            String tenant = request.getHeader(HEADER_TENANT);
+            publishEvent(request, response, tenant, null, null);
+        } else {
+            String userLogin = auth.getLogin().orElse("");
+            String userKey = auth.getUserKey().orElse("");
+            String tenant = auth.getTenantName().orElse("");
+
+            publishEvent(request, response, tenant, userLogin, userKey);
+        }
+    }
+
+    // TODO
+    private boolean isIgnoredRequest(HttpServletRequest request, HttpServletResponse response) {
+        return false;
+    }
+
+    private void publishEvent(HttpServletRequest request, HttpServletResponse response, String tenant, String userLogin, String userKey) {
+        DomainEvent domainEvent = DomainEvent.builder()
+            .id(UUID.randomUUID())
+            .tenant(tenant)
+            .userKey(userKey)
+            .payload(createPayload(request, response))
+            .build();
+        eventPublisher.publish(DefaultDomainEventSource.HTTP.name(), domainEvent);
+    }
+
+    @SneakyThrows
+    private HttpDomainEventPayload createPayload(HttpServletRequest request, HttpServletResponse response) {
+        ContentCachingResponseWrapper responseCacheWrapperObject = new ContentCachingResponseWrapper(response);
+
+        String requestBody = IOUtils.toString(request.getReader());
+        String responseBody = IOUtils.toString(responseCacheWrapperObject.getContentInputStream(), StandardCharsets.UTF_8);
+
+        HttpDomainEventPayload payload = new HttpDomainEventPayload();
+        payload.setMethod(request.getMethod());
+        payload.setUrl(request.getRequestURI());
+        payload.setQueryString(request.getQueryString());
+        payload.setRequestLength(request.getContentLengthLong());
+        payload.setRequestBody(requestBody);
+        payload.setResponseBody(responseBody);
+        payload.setResponseLength((long) responseBody.length());
+        payload.setRequestHeaders(getRequestHeaders(request));
+        payload.setResponseHeaders(getResponseHeaders(response));
+        payload.setResponseCode(response.getStatus());
+        payload.setExecTime(MdcUtils.getExecTimeMs());
+        return payload;
+    }
+
+    private HttpHeaders getRequestHeaders(HttpServletRequest httpRequest) {
+        Map<String, List<String>> headersMap = Collections.list(httpRequest.getHeaderNames())
+            .stream()
+            .collect(Collectors.toMap(
+                Function.identity(),
+                h -> Collections.list(httpRequest.getHeaders(h))
+            ));
+
+        return HttpHeaders.of(headersMap, (s1, s2) -> true);
+    }
+
+    private HttpHeaders getResponseHeaders(HttpServletResponse httpServletResponse) {
+        Map<String, List<String>> headersMap = httpServletResponse.getHeaderNames().stream()
```

Comment:
please check current logic of header processing:
com.icthh.xm.commons.timeline.TimelineEventProducer#getRequestHeaders